### PR TITLE
preparation for application to the YJ band

### PR DESCRIPTION
**Pull request by YuiKasagi**
Some instance variables have been added to treat YJ band or mmf1 (comb fiber) data. (See IRD_stream.py for usage.)

memo:
** The reduction process is NOT YET optimized for YJ band data.
**  The problem of setting the appropriate path for the data (e.g. the reference path of the master ThAr file) is NOT YET resolved.
The above things will be resolved in the future.

### CODE REVIEW

**Review by HajimeKawahara — APPROVED** [review]
All of the unit tests passed. Thanks!

File: src/pyird/utils/irdstream.py [hunk 1]
```diff
@@ -370,44 +382,49 @@ def mkwspec(spec_m2,reference,save_path):
...
+                save_path = self.anadir/('%s%d_%s.dat'%(prefix,id,self.trace.mmf)) ##e.g. w41511_m2.dat
                 wspec = mkwspec(spec_m2,reference,save_path)
                 if self.info:
-                    print('dispcor: output spectrum= %s%d_m2.dat'%(prefix,id))
+                    print('dispcor: output spectrum= %s%d_%s.dat'%(prefix,id,self.trace.mmf))
                 #plot
                 show_wavcal_spectrum(wspec,alpha=0.5)
         else:
-            hdu = pyf.open(self.anadir/('%s_mmf12.fits'%(self.streamid)))[0]
+            hdu = pyf.open(self.anadir/('%s_mmf12_%s.fits'%(self.streamid,self.band)))[0]
```
Comment (kemasuda) [inline src/pyird/utils/irdstream.py:392]: self.trace.mmf?

**Review by kemasuda — APPROVED** [review]
Thanks for this update, the PR passed all the tests in my environment.

I have one question about the updated naming convention. In IRD_stream.py, the flattened flat (flat_mmf12_h.fits) has the tag mmf12, although it seems to be flatted using mmf2 aperture. Is this intentional?

File: src/pyird/utils/irdstream.py (hunk 1 above)
Comment (YuiKasagi) [reply src/pyird/utils/irdstream.py:392]: Thank you for pointing this out!! I just missed changing this part, so it was not intentional.
I will fix the naming convention to {streamID}_{band}_{mmf}.fits for the master file.

**Comment by YuiKasagi** [comment]
Thank you for your review! I made slight modifications.
Could you check it again? @HajimeKawahara  @kemasuda

**Comment by kemasuda** [comment]
Thanks for checking, looks good to me!

File: examples/python/IRD_stream.py [hunk 2]
```diff
@@ -25,21 +33,31 @@
...
+elif flat.band=='y':
+    rawtag='IRDBD000'
 
 #wavelength calibration
-thar=irdstream.Stream2D("thar",datadir,anadir,rawtag="IRDAD000",fitsid=list(range(14632,14732)))
+thar=irdstream.Stream2D("thar",datadir,anadir,rawtag=rawtag,fitsid=list(range(14632,14732)))
 thar.trace = trace_mmf
-trace_mmf.mmf2() # choose mmf2 (star fiber)
 thar.clean_pattern(extin='', extout='_cp', hotpix_mask=hotpix_mask)
 thar.calibrate_wavlength()
```
Comment (HajimeKawahara) [inline examples/python/IRD_stream.py:62]: calibrate_wavelength?

Comment (YuiKasagi) [reply examples/python/IRD_stream.py]: That's right... I fixed it.

**Comment by HajimeKawahara** [comment]
Thanks!

**Pull request merged**